api/src/main/java/jakarta/data/page/Page.java

```diff
@@ -146,17 +146,33 @@ default Stream<T> stream() {
      * Returns the total number of elements across all pages of query results, if the
      * {@link #pageRequest()} specified that {@linkplain PageRequest#requestTotal the
      * total should be retrieved from the database}.
+     *
+     * <p><strong>Note for NoSQL databases:</strong> Not all NoSQL databases support counting the total
+     * number of elements. This operation is <strong>not supported</strong> for Key-Value and Wide-Column databases.
+     * For Graph and Document databases, support for this operation may vary depending on the provider.
+     * If the database does not support retrieving the total number of elements, calling this method will
+     * result in an {@link UnsupportedOperationException}.</p>
+     *
      * @return the total number of elements across all pages.
      * @throws IllegalStateException if the total was not retrieved from the database.
+     * @throws UnsupportedOperationException if the database does not support total element queries.
```

Comment:
It seems confusing to me to add UnsupportedOperationException alongside the existing IllegalStateException which the method already throws "if the total was not retrieved from the database."  I propose that we don't add the new exception or any of the preceding language and just add a sentence to the end of the exception we already have instead:

```suggestion
     * @throws IllegalStateException if the total was not retrieved from the database.
     *                               Some NoSQL databases are not capable of retrieving
     *                               a total number of elements.
```

Reply:
But @njr-11 IMHO: it is a different scenario, I mean: 

- IllegalStateException: if the total was not retrieved from the database.

The point is there are NoSQL databases that cannot do it; IMHO, in this scenario, UnsupportedOperationException fits semantically better.